measures/Measures/MCFrame.cc

```diff
@@ -487,114 +508,117 @@ void MCFrame::makeEpoch() {
+  std::unique_ptr<casacore::Measure>& epoch = myf.lock()->epval;
+  epConvTDB = new MEpoch::Convert(*epoch, REFTDB);
+  epConvUT1 = new MEpoch::Convert(*epoch, REFUT1);
+  epConvTT  = new MEpoch::Convert(*epoch, REFTT);
   if (epTDBp) {
-    delete epTDBp; epTDBp = 0;
+    delete epTDBp; epTDBp = nullptr;
   }
   if (epUT1p) {
-    delete epUT1p; epUT1p = 0;
+    delete epUT1p; epUT1p = nullptr;
   }
   if (epTTp) {
-    delete epTTp; epTTp = 0;
+    delete epTTp; epTTp = nullptr;
   }
   if (epConvLAST) {
     delete static_cast<MEpoch::Convert *>(epConvLAST);
-    epConvLAST = 0;
+    epConvLAST = nullptr;
   }
-  epConvLAST = new MEpoch::Convert(*(myf.epoch()),
-				   MEpoch::Ref(MEpoch::LAST, this->myf));
+  epConvLAST = new MEpoch::Convert(*(epoch),
+				   MEpoch::Ref(MEpoch::LAST, MeasFrame(this->myf.lock())));
   if (epLASTp) {
-    delete epLASTp; epLASTp = 0;
+    delete epLASTp; epLASTp = nullptr;
   }
   if (appLongp) {
-    delete appLongp; appLongp = 0;
-    delete dirAppp; dirAppp = 0;
+    delete appLongp; appLongp = nullptr;
+    delete dirAppp; dirAppp = nullptr;
   }
   if (radLSRp) {
-    delete radLSRp; radLSRp = 0;
+    delete radLSRp; radLSRp = nullptr;
   }
 }
 
 void MCFrame::makePosition() {
   static const MPosition::Ref REFLONG 
     = MPosition::Ref(MPosition::ITRF);
   delete static_cast<MPosition::Convert *>(posConvLong);
-  posConvLong = new MPosition::Convert(*(myf.position()),
+  std::unique_ptr<casacore::Measure>& position = myf.lock()->posval;
+  posConvLong = new MPosition::Convert(*position,
 				       REFLONG);
   if (posLongp) {
-    delete posLongp; posLongp = 0;
-    delete posITRFp; posITRFp = 0;
+    delete posLongp; posLongp = nullptr;
+    delete posITRFp; posITRFp = nullptr;
   }
   if (epLASTp) {
-    delete epLASTp; epLASTp = 0;
+    delete epLASTp; epLASTp = nullptr;
   }
   if (radLSRp) {
-    delete radLSRp; radLSRp = 0;
+    delete radLSRp; radLSRp = nullptr;
   }
   static const MPosition::Ref REFGEO
     = MPosition::Ref(MPosition::WGS84);
   delete static_cast<MPosition::Convert *>(posConvLongGeo);
-  posConvLongGeo = new MPosition::Convert(*(myf.position()),
+  posConvLongGeo = new MPosition::Convert(*position,
 					  REFGEO);
   if (posLongGeop) {
-    delete posLongGeop; posLongGeop = 0;
-    delete posGeop; posGeop = 0;
+    delete posLongGeop; posLongGeop = nullptr;
+    delete posGeop; posGeop = nullptr;
   }
 }
 
 void MCFrame::makeDirection() {
   static const MDirection::Ref REFJ2000 = MDirection::Ref(MDirection::J2000);
-  uInt locker =0;
   if (dirConvJ2000) {
     delete static_cast<MDirection::Convert *>(dirConvJ2000);
-    dirConvJ2000 = 0;
+    dirConvJ2000 = nullptr;
   }
-  dirConvJ2000 = new MDirection::Convert(*(myf.direction()),
+  std::unique_ptr<casacore::Measure>& direction = myf.lock()->dirval;
+  dirConvJ2000 = new MDirection::Convert(*direction,
 					 MDirection::Ref(MDirection::J2000,
-							 this->myf));
+							 MeasFrame(this->myf.lock())));
```

Comment:
I think this is the key of the failure in my patch. If I add this then the test that was failing is now working. Great!

Reply:
I thought what went wrong in your case was this scenario:
- Make a MeasFrame A (creates a Rep, let's say R. R points via a MCFrame to A.)
- Copy MeasFrame A to B. B points also to R, R points via a MCFrame to A.
- Destruct A. R still exists because B points to it. R points to non-existing A.
That scenario I tackled by letting MCFrame point directly to R instead of to a MeasFrame.